src/mjolnir/shortcutbuilder.cc

```diff
@@ -81,21 +81,25 @@ bool EdgesMatch(const graph_tile_ptr& tile, const DirectedEdge* edge1, const Dir
   // TODO - should allow near matches?
   std::vector<std::string> edge1names = tile->GetNames(edge1);
   std::vector<std::string> edge2names = tile->GetNames(edge2);
-  if (edge1names.size() != edge2names.size()) {
+  std::sort(edge1names.begin(), edge1names.end());
+  std::sort(edge2names.begin(), edge2names.end());
+  if (edge1names != edge2names)
     return false;
-  }
-  for (const auto& name1 : edge1names) {
-    bool found = false;
-    for (const auto& name2 : edge2names) {
-      if (name1 == name2) {
-        found = true;
-        break;
-      }
-    }
-    if (!found) {
+
+  // if they have access restrictions those must match (for modes that use shortcuts)
+  if (edge1->access_restriction()) {
+    auto res1 = tile->GetAccessRestrictions(edge1 - tile->directededge(0), kVehicularAccess);
+    auto res2 = tile->GetAccessRestrictions(edge2 - tile->directededge(0), kVehicularAccess);
+    if (res1.size() != res2.size())
       return false;
+    for (size_t i = 0; i < res1.size(); ++i) {
+      res1[i].set_edgeindex(0);
+      res2[i].set_edgeindex(0);
+      if (res1[i].value() != res2[i].value())
+        return false;
```

Comment:
right, I kinda forgot you already had an active PR @kevinkreiser . since those restrictions are 90% truck specific (+ time-based restrictions), I wonder if we shouldn't make this somehow configurable? it seems like a worse impact than allowing tunnels to break up shortcuts..

Reply:
on first instinct, i dont like the idea of making it configurable. if you can configure it to give you incorrect routes we just leave the door open to all kinds of comments about: hey my routes are broken. instead, id rather break shortcuts everywhere where its necessary and measure performance differences. if its a huge performance killer for sure then we should start working on an alternative. my 2 cents

Reply:
an example alternative would be to stiil build the shortcuts for modes that could use it but exclude trucks from using the shortcut. havent thought it all the way through and it surely makes shortcut building suck but hey might be worth it if it keep auto costing perf up